src/Mesh/Interpolation.jl

```diff
@@ -0,0 +1,442 @@
+            lag    = @view sv[:,st_no,el]
+
+            phir = Elements.interpolationmatrix(m1_r, intrp_brck.ξ1[el], wb)
+            phis = Elements.interpolationmatrix(m1_r, intrp_brck.ξ2[el], wb)
+
+            for k in 1:l3 # interpolating point-by-point
+                interpolationmatrix_1pt(m1_r,intrp_brck.ξ3[el][k],wb,phit)
+
+                for j in 1:l2, i in 1:l1
+                    vout = 0.0            
+                    for ik in 1:qm1
+                        v_ij = 0.0
+                        for ij in 1:qm1
+                            v_ii = 0.0
+                            for ii in 1:qm1
+                                @inbounds v_ii += lag[ii + (ij-1)*qm1 + (ik-1)*qm1*qm1] * phir[i,ii]
+                            end # ii loop
+
+                            @inbounds v_ij += v_ii * phis[j,ij]
+                        end # ij loop
+                        @inbounds vout += v_ij * phit[ik]
+                    end # ik loop
+                    @inbounds intrp_brck.V[el][i + (j-1)*l1 + (k-1)*l1*l2] = vout 
+                end # j, i loop
+            end # k loop
+        end
+  #--------------------
+  end
+end
+#--------------------------------------------------------
+function interpolate_brick_v2!(intrp_brck::InterpolationBrick, sv::AbstractArray{FT}, st_no::T, poly_order::T) where {T <: Integer, FT <: AbstractFloat}
+    qm1 = poly_order + 1
+    Nel = length(intrp_brck.realelems)
+    m1_r, m1_w = GaussQuadrature.legendre(FT,qm1,GaussQuadrature.both)
+    wb = Elements.baryweights(m1_r)
+    Nel = length(intrp_brck.realelems)
+    alpha = FT(1)
+    beta  = FT(0)
+    #-----for each element elno 
+    for el in 1:Nel
+        if length(intrp_brck.ξ1[el]) > 0
+            lag    = @view sv[:,st_no,el]
+            phir = Elements.interpolationmatrix(m1_r, intrp_brck.ξ1[el], wb)
+            phis = Elements.interpolationmatrix(m1_r, intrp_brck.ξ2[el], wb)
+            phit = Elements.interpolationmatrix(m1_r, intrp_brck.ξ3[el], wb)
+
+            sr,si = size(phir); ss,sj = size(phis); st,sk = size(phit)
+            #---------------------------------------------------------------
+            h1 = zeros(FT, sj*sk, sr)  # this is for testing only
+            h2 = zeros(FT, sk*sr, ss)  # memory allocations within the function will be removed
+            #---------------------------------------------------------------
+            LinearAlgebra.BLAS.gemm!('T', 'T', alpha, reshape(lag,si, sj*sk), phir, beta, h1)
+            LinearAlgebra.BLAS.gemm!('T', 'T', alpha, reshape(h1, sj, sk*sr), phis, beta, h2)
+            LinearAlgebra.BLAS.gemm!('T', 'T', alpha, reshape(h2, sk, sr*ss), phit, beta, reshape(intrp_brck.V[el], sr*ss, st))
+            #---------------------------------------------------------------
+        end
+  #--------------------
+  end
+end
+#--------------------------------------------------------
+@inline function interpolationmatrix_1pt(rsrc::Vector{FT}, rdst::FT,
```

Comment:
it doesn't return a matrix, so maybe `interpolation_weights`?

Also would be good to add docs.